soh/src/code/z_message_PAL.c

```diff
@@ -971,33 +1000,34 @@ void Message_DrawText(PlayState* play, Gfx** gfxP) {
                 }
                 *gfxP = gfx;
                 return;
-            case MESSAGE_SHIFT:
-                msgCtx->textPosX += msgCtx->msgBufDecoded[++i];
+            case MESSAGE_SHIFT_JPN:
+                msgCtx->textPosX += msgCtx->msgBufDecodedWide[++i];
                 break;
-            case MESSAGE_TEXTID:
+            case MESSAGE_TEXTID_JPN:
+                // #region SOH [General] Fixes softlock for higher text speeds
+                if (gTextSpeed > 1) {
+                    msgCtx->textDrawPos = i + 1;
+                }
+                // #endregion
                 msgCtx->textboxEndType = TEXTBOX_ENDTYPE_HAS_NEXT;
                 if (msgCtx->msgMode == MSGMODE_TEXT_DISPLAYING) {
                     Audio_PlaySoundGeneral(0, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale, &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
                     msgCtx->msgMode = MSGMODE_TEXT_DONE;
-                    Font_LoadMessageBoxIcon(font, TEXTBOX_ICON_TRIANGLE);
+                    Font_LoadMessageBoxIcon(&play->msgCtx.font, TEXTBOX_ICON_TRIANGLE);
                 }
                 *gfxP = gfx;
                 return;
-            case MESSAGE_QUICKTEXT_ENABLE:
+            case MESSAGE_QUICKTEXT_ENABLE_JPN:
                 if (i < msgCtx->textDrawPos && i + gTextSpeed >= msgCtx->textDrawPos && (msgCtx->msgMode == MSGMODE_TEXT_DISPLAYING ||
```

Comment:
Looks like you changed the decomp portion of this for somereason. I think this should be reverted and maybe the condition changed to `||` the fast text part.
```suggestion
                if ((i + 1 == msgCtx->textDrawPos || (gTextSpeed > 1 && i + gTextSpeed >= msgCtx->textDrawPos)) && (msgCtx->msgMode == MSGMODE_TEXT_DISPLAYING ||
```

Reply:
I made it match what we have in our english section, i agree though that I don't like how it is written (since its just unclear that its the same thing when gTextSpeed = 1) so will change both sections to something like this